rpcs3/Emu/System.cpp

```diff
@@ -512,21 +532,17 @@ bool Emulator::BootGame(const std::string& path, const std::string& title_id, bo
 	if (direct && fs::exists(path))
 	{
 		m_path = path;
-		Load(title_id, add_only, force_global_config);
-		return true;
+		return Load(title_id, add_only, force_global_config);
 	}
 
-	bool success = false;
 	for (std::string elf : boot_list)
 	{
 		elf = path + elf;
 
 		if (fs::is_file(elf))
 		{
 			m_path = elf;
-			Load(title_id, add_only, force_global_config);
-			success = true;
-			break;
+			return Load(title_id, add_only, force_global_config);
```

Comment:
You are completely breaking the logic here.
BootGame is not supposed to return before reaching the endof the function.